chore(docs): Add Stack documentation

BenJamesBen opened the pull request:
I am especially requesting that @joostdecock and @eriese look at this PR. This is because I did not understand stacks before starting to write this documentation. I had to learn as I went along.

BenJamesBen commented:
This partially covers #3232.

eriese commented on markdown/dev/guides/patterns/stacks/en.md:
```diff
@@ -3,7 +3,7 @@ title: Stacks
-Stacks come into play when laying out a pattern.
+[Stacks](/reference/api/stack) come into play when laying out a pattern.
 The FreeSewing core library, by default, will handle the layout of a pattern
 for you by placing all parts next to each other in as small a space as
 possible.
```
Let's change this to "by placing all parts into stacks, and arranging the stacks in as small a space as possible without overlapping". 

The point is that parts in the same stack will overlap, but stacks shouldn't because they are the boxes used by the layout engine

BenJamesBen replied:
Added.

eriese commented on markdown/dev/guides/patterns/stacks/en.md:
```diff
@@ -46,12 +46,21 @@ to patterns designers who want to use it.
 
 Essentially, stacks behave as layers. Parts that are on the same _stack_ will be stacked on top of each other in the layout.
```
I think this sentence actually confuses the matter. possibly an improvement: "Essentially a stack is a collection of parts. Parts in a stack act as layers, and will be stacked directly on top of each other."

BenJamesBen replied:
Added.

eriese commented on markdown/dev/reference/api/part/config/stack/en.md:
```diff
@@ -0,0 +1,45 @@
+This is because parts in the stack are drafted within the same stack
+space.
+
+Otherwise, if the `stack` property is not present, the default behavior
+is to assign the part to its own stack.
```
Otherwise, if the `stack` property is not present, the default behavior
is to use the part's name as its stack name.

BenJamesBen replied:
Added.

eriese commented on markdown/dev/reference/api/part/config/stack/en.md:
```diff
@@ -0,0 +1,45 @@
+
+Otherwise, if the `stack` property is not present, the default behavior
+is to assign the part to its own stack.
+The part will not overlap other parts in drafting and printing layouts.
+This is because it is the only part drafted within its stack and
+because stacks do not overlap other stacks.
```
In a draft with only one set, this will result in each part having its own stack. With a default layout, the part will not overlap other parts because it is the only part drafted within its stack and stacks do not overlap.

In a draft with multiple sets, this will result in parts of the same name using the same stack. This is how we achieve the layered look of parts in sample drafts.

BenJamesBen replied:
Added.

eriese commented on markdown/dev/reference/api/stack/asProps/en.md:
````diff
@@ -0,0 +1,12 @@
+
+```js
+Object stack.asProps()
+```
````
I think this wants a `Returned Object` section, like in [Patter.getRenderProps()](https://github.com/freesewing/freesewing/blob/develop/markdown/dev/reference/api/pattern/getrenderprops/en.md#patterngetrenderprops-returned-object)

The `Stack.asProps()` method returns an object with 
the following properties:

| Property | Description |
| --------:| ----------- |
| `attributes` | An [`Attributes`](/reference/api/attributes) instance holding the stack's attributes |
| `bottomRight` | A [`Point`](/reference/api/point) that is the bottom right of the stack's bounding box |
| `height` | Height of the bounding box of the stack in `mm` |
| `name` | The name of the stack |
| `parts` | An `Array` of the `Part`s in the stack |
| `topLeft` | A [`Point`](/reference/api/point) that is the top left of the stack's bounding box |
| `width` | Width of the bounding box of the stack in `mm` |

BenJamesBen replied:
Added.

eriese commented on markdown/dev/reference/api/stack/en.md:
```diff
@@ -0,0 +1,52 @@
+title: Stack
+---
+
+A Stack object represents a layer within a pattern, holding one or
```
See above about the stacks-as-layers thing

BenJamesBen replied:
Changed to:
A Stack object represents a collection of parts within a pattern. Stacks are used when laying out the pattern.

eriese commented on markdown/dev/reference/api/stack/en.md:
```diff
@@ -0,0 +1,52 @@
+
+Stack objects come with the following properties:
+
+- `attributes` : An [Attributes](/reference/api/attributes) instance holding the point's attributes
```
stack's attributes

BenJamesBen replied:
Fixed.

eriese commented on markdown/dev/reference/api/stack/generateTransform/en.md:
```diff
@@ -0,0 +1,26 @@
+title: Stack.generateTransform()
+---
+
+The `Stack.generateTransform()` method generates transforms for the stack,
```
let's say "generates SVG transforms for the stack"

BenJamesBen replied:
Added.

eriese commented on markdown/dev/reference/api/stack/home/en.md:
```diff
@@ -0,0 +1,17 @@
+title: Stack.home()
+---
+
+The `Stack.home()` method calculates the stack's bounding box, sets it,
```
change "sets it" to, "setting the `width`, `height`, `topLeft` and `bottomRight` properties",

BenJamesBen replied:
Changed as suggested.

eriese commented on markdown/dev/reference/api/stack/home/en.md:
```diff
@@ -0,0 +1,17 @@
+
+## Notes
+
+The bounding box is calculated taking into consideration the `margin` setting.
```
another note:
`Stack.home()` caches its calculations by checking whether the stack's `topLeft` property has been set. If `topLeft` has a truthy value, it will not re-calculate.

BenJamesBen replied:
Added.

eriese reviewed (changes requested):
Really great job! I just made a few clarification notes. If they're not helpful or need different wording I trust your judgment

BenJamesBen commented on markdown/dev/guides/patterns/stacks/en.md:
```diff
@@ -46,12 +46,21 @@ to patterns designers who want to use it.
 In the vast majority of cases you won't be using any stacks, or the stacking
 will be handled for you by the core library (like in the sampling example
 above).
+
```
Added.

eriese commented:
Well, I took too long to approve this, and now apparently there's a conflict. @BenJamesBen can you please update as needed?

BenJamesBen commented:
I reverted the Pattern Guide file changes, rebased to `develop`, applied the edits to the new Design Guide file, and edited two other files to correct the URL to point to the new Design Guide.
